File: src/main/java/com/pauljoda/modularsystems/storage/tiles/TileStorageCapacity.java
package com.pauljoda.modularsystems.storage.tiles;

import net.minecraft.entity.item.EntityItem;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;

import java.util.List;
import java.util.Random;

/**
 * Modular-Systems
 * Created by Paul Davis on 7/24/2015
 */
public class TileStorageCapacity extends TileStorageBasic {
    @Override
    public void addedToNetwork() {
        super.addedToNetwork();
        getCore().pushNewInventory(11);
        worldObj.markBlockForUpdate(core.x, core.y, core.z);
    }

    @Override
    public void removedFromNetwork() {
        super.removedFromNetwork();
        if(getCore() != null) {
            List<ItemStack> stacks = getCore().popInventory(11);
            for (ItemStack itemStack : stacks) {
                Random rand = new Random();
                if (itemStack != null && itemStack.stackSize > 0) {
                    float rx = rand.nextFloat() * 0.8F + 0.1F;
                    float ry = rand.nextFloat() * 0.8F + 0.1F;
                    float rz = rand.nextFloat() * 0.8F + 0.1F;

                    EntityItem entityItem = new EntityItem(worldObj,
                            xCoord + rx, yCoord + ry, zCoord + rz,
                            new ItemStack(itemStack.getItem(), itemStack.stackSize, itemStack.getItemDamage()));

                    if (itemStack.hasTagCompound())
                        entityItem.getEntityItem().setTagCompound((NBTTagCompound) itemStack.getTagCompound().copy());

                    float factor = 0.05F;
                    entityItem.motionX = rand.nextGaussian() * factor;
                    entityItem.motionY = rand.nextGaussian() * factor + 0.2F;
                    entityItem.motionZ = rand.nextGaussian() * factor;
                    worldObj.spawnEntityInWorld(entityItem);

                    itemStack.stackSize = 0;
                }
            }
            worldObj.markBlockForUpdate(core.x, core.y, core.z);
        }
    }
}

File: src/main/scala/com/teambr/modularsystems/storage/tiles/TileStorageCapacity.scala
package com.teambr.modularsystems.storage.tiles

import java.util.Random

import net.minecraft.entity.item.EntityItem
import net.minecraft.item.ItemStack
import net.minecraft.nbt.NBTTagCompound

/**
 * Modular-Systems
 * Created by Dyonovan on 06/08/15
 */
class TileStorageCapacity extends TileEntityStorageExpansion {

    override def addedToNetwork(): Unit = {
        super.addedToNetwork()
        getCore.get.addInventorySlots(11)
        worldObj.markBlockForUpdate(getCore.get.getPos)
    }

    override def removedFromNetwork(): Unit = {
        super.removedFromNetwork()
        getCore match {
            case Some(theCore) =>
            val itemStacks = theCore.removeInventorySlots(11)
            for (i <- itemStacks.indices) {
                val rand: Random = new Random
                if (itemStacks(i) != null && itemStacks(i).stackSize > 0) {
                    val rx: Float = rand.nextFloat * 0.8F + 0.1F
                    val ry: Float = rand.nextFloat * 0.8F + 0.1F
                    val rz: Float = rand.nextFloat * 0.8F + 0.1F
                    val entityItem: EntityItem = new EntityItem(worldObj, pos.getX + rx, pos.getY + ry, pos.getZ + rz,
                        new ItemStack(itemStacks(i).getItem, itemStacks(i).stackSize, itemStacks(i).getItemDamage))
                    if (itemStacks(i).hasTagCompound) entityItem.getEntityItem.setTagCompound(itemStacks(i).
                            getTagCompound.copy.asInstanceOf[NBTTagCompound])
                    val factor: Float = 0.05F
                    entityItem.motionX = rand.nextGaussian * factor
                    entityItem.motionY = rand.nextGaussian * factor + 0.2F
                    entityItem.motionZ = rand.nextGaussian * factor
                    worldObj.spawnEntityInWorld(entityItem)
                    itemStacks(i).stackSize = 0
                }
            }
            worldObj.markBlockForUpdate(getCore.get.getPos)
            case _ =>
        }
    }
}
